src/test/kotlin/sirius/biz/analytics/events/EventRecorderTest.kt

```diff
@@ -0,0 +1,118 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.biz.analytics.events
+
+import org.junit.jupiter.api.AfterAll
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import sirius.db.jdbc.Databases
+import sirius.kernel.SiriusExtension
+import sirius.kernel.async.BackgroundLoop
+import sirius.kernel.di.std.Part
+import java.time.Duration
+import java.util.*
+import java.util.concurrent.atomic.AtomicInteger
+import kotlin.test.assertEquals
+import kotlin.test.assertNull
+
+@ExtendWith(SiriusExtension::class)
+class EventRecorderTest {
+
+    companion object {
+        @Part
+        private lateinit var recorder: EventRecorder
+
+        @Part
+        private lateinit var dbs: Databases
+
+        @BeforeAll
+        @JvmStatic
+        fun setup() {
+            BackgroundLoop.disable(EventProcessorLoop::class.java).await(Duration.ofMinutes(1))
+        }
+
+        @AfterAll
+        @JvmStatic
+        fun cleanup() {
+            BackgroundLoop.enable(EventProcessorLoop::class.java)
+        }
+    }
+
+    @Test
+    fun `Events are recorded`() {
+        recorder.record(TestEvent1())
+        recorder.record(TestEvent1())
+        recorder.record(TestEvent1())
+        recorder.record(TestEvent1())
+
+        recorder.record(TestEvent2())
+        recorder.record(TestEvent2())
+        recorder.record(TestEvent2())
+        recorder.record(TestEvent2())
+
+        recorder.process()
+
+        assertEquals(4, dbs.get("clickhouse").createQuery("SELECT * FROM testevent1").queryList().size)
+        assertEquals(4, dbs.get("clickhouse").createQuery("SELECT * FROM testevent2").queryList().size)
+    }
+
+    @Test
+    fun `BufferedEvents-field is reset after processing`() {
+        recorder.record(TestEvent1())
+        recorder.record(TestEvent1())
+        recorder.record(TestEvent1())
+        recorder.record(TestEvent1())
+
+        recorder.record(TestEvent2())
+        recorder.record(TestEvent2())
+        recorder.record(TestEvent2())
+        recorder.record(TestEvent2())
+
+        recorder.javaClass.getDeclaredField("bufferedEvents").apply {
```

Comment:
Komische Frage vom Kotlin-Noob – Warum ist das im Vergleich zu Groovy komplizierter?

Reply:
Because Kotlin, like Java, handles visibility properly and `bufferedEvents` is a private field. With Groovy you can access everything and nothing (even non-existent methods, ...), with Java and Kotlin only what an external class in the same package could access.

An alternative solution would be declare the field as `protected`.

Reply:
Ah💡 ... Thanks for the clarification!